acceptance/support/application-manager.ts

```diff
@@ -0,0 +1,88 @@
+import * as fs from "fs";
+import {Logger} from "tslog";
+import {uploadPath} from "./init";
+import {TestData} from "./test-constants";
+
+/**
+ * Utilities for managing LSRS while executing the Cucumber acceptance tests.
+ */
+export default class ApplicationManager {
+
+    private static readonly startUpTimeout = 5000;
+    private static readonly logger: Logger = new Logger({
+        name: "Cucumber | ApplicationManager",
+        minLevel: "debug",
+        displayFunctionName: false,
+        displayFilePath: "hidden"
+    });
+
+    private static started: boolean = false;
+
+    /**
+     * Starts up LSRS. Returns immediately if the application is already running.
+     */
+    public static async start(): Promise<void> {
+
+        return new Promise((resolve) => {
+
+            if (this.started) {
+                this.notifyStarted(resolve, "Application is already loaded");
+            } else {
+
+                this.logger.info(`Starting application with environment=${process.env.NODE_ENV}...`);
+
+                this.started = true;
+                import("../../src/lsrs-main");
+
+                setTimeout(() => {
+                    this.prepareMockStorage();
+                    this.notifyStarted(resolve, "Assuming application is now running");
+                }, this.startUpTimeout);
+            }
+        });
+    }
+
+    /**
+     * Creates the required mock folders and files for test execution.
+     */
+    public static prepareMockStorage() {
+
+        this.clearAcceptorRoot("images");
+        this.clearAcceptorRoot("files");
+
+        this.createTestFiles(TestData.subFolders, fs.mkdirSync);
+        this.logger.info("Created mock storage folders");
+
+        this.createTestFiles(TestData.files, (filename) => fs.writeFileSync(filename, Buffer.from(filename)));
+        this.logger.info("Created mock files");
+    }
+
+    /**
+     * Removes the temporary storage folder.
+     */
+    public static cleanUp(basePath = uploadPath): void {
+
+        fs.rmSync(basePath, {recursive: true, force: true});
+        this.logger.info(`Removed temporary storage folder (${basePath})`);
+    }
+
+    private static clearAcceptorRoot(acceptor: string) {
+
+        fs.readdirSync(`${uploadPath}/${acceptor}`)
+            .map(subFolder => `${uploadPath}/${acceptor}/${subFolder}`)
+            .forEach(absolutePath => this.cleanUp(absolutePath));
+    }
+
+    private static notifyStarted(resolve: () => void, message: string) {
+        this.logger.info(message);
+        resolve();
+    }
+
+    private static createTestFiles(items: string[], factoryFunction: (path: string) => void) {
```

Comment:
Personally I would separate this method into two. I see they share parts, but that last param can be confusing on first read. Or maybe just "hide" this method behind two other methods with more descriptive names.
If you only need files to be created ( i.e. no empty directories ), you could also just introduce an extra mkdir call that creates the parent directory first.

Reply:
I need some empty dirs as well, but the other is a fair point, I'll split the method.